kpatch-build/lookup.c

```diff
@@ -178,19 +178,25 @@ static void find_local_syms(struct lookup_table *table, char *hint,
 /* Strip the path and replace '-' with '_' */
 static char *make_modname(char *modname)
 {
-	char *cur;
+	char *cur, *name;
 
 	if (!modname)
 		return NULL;
 
-	cur = modname;
+	cur = strdup(modname);
+	if (!cur)
+		perror("strdup");
+
 	while (*cur != '\0') {
 		if (*cur == '-')
 			*cur = '_';
 		cur++;
 	}
 
-	return basename(modname);
+	name = basename(cur);
+	name = strdup(name);
+	free(cur);
```

Comment:
Sorry, I should have done a careful review before, the code segfaults at `free (cur)`.  Segfault happens because the string start position (length) of `cur` is different while allocation vs free'ing,  that also mean the basename is wrong.  I was curious and checking the memory saving, with and without the patch:
```
Before:
==12851== LEAK SUMMARY:
==12851==    definitely lost: 2,257,092 bytes in 94,142 blocks
==12851==    indirectly lost: 6,044,367 bytes in 114,647 blocks

After: (fixing the segfault by fixing cur using tmp variable)
==13365==    definitely lost: 2,257,060 bytes in 94,141 blocks
==13365==    indirectly lost: 454,869 bytes in 74 blocks
```
This patch does fixes huge amount memory leak. I guess the memory leak seen after the patch due to the un-freed allocation from other paths.   One way to fix is, the segfault:
```
@@ -187,12 +187,14 @@ static char *make_modname(char *modname)
        if (!cur)
                perror("strdup");
 
+       name = cur; /* use name as tmp */
        while (*cur != '\0') {
                if (*cur == '-')
                        *cur = '_';
                cur++;
        }
 
+       cur = name;
        name = basename(cur);
        name = strdup(name);
        free(cur);
```

Reply:
Sorry， I am careless, thank you for your careful review.

Reply:
Sorry, I thought I left this comment earlier, but it got lost somehow -- maybe I forgot to submit it.

I think this code could be simplified a bit, like:
```
	name = strdup(basename(modname));
	if (!name)
		perror("strdup");

	cur = name;
	while (*cur != '\0') {
		if (*cur == '-')
			*cur = '_';
		cur++;
	}

	return name;
```

Reply:
OK, it's good. Thanks！